packages/docs-ui/src/services/selection/doc-selection-render.service.ts

```diff
@@ -176,17 +177,39 @@ export class DocSelectionRenderService extends RxDisposable implements IRenderMo
             const { startOffset, endOffset, rangeType } = range;
 
             if (rangeType === DOC_RANGE_TYPE.RECT) {
-                const rectRange = getRectRangeFromCharIndex(
-                    startOffset, endOffset, scene, document, docSkeleton, style, segmentId, segmentPage
-                );
+                const { startNodePosition, endNodePosition } = range as ITextRangeWithStyle;
```

Comment:
Code duplications.

Reply:
The code is not the same here, one is to create TextRanges and the other is a RectRange, and it is unlikely that it will go towards the same code in the future